### PR TITLE
Add support for okhttpVersion, okioVersion and reactNativeVersion gradle root-project extra properties. 

### PR DESCRIPTION
**Pull request by jgreen210**
- [ ] This is a small change 
- [ ] This change has been discussed in issue  #1380  and the solution has been agreed upon with maintainers.

---

**Description:**

The aim is to allow version conflicts to be overridden if override the kotlin version. It's also following the standard pattern that most other projects use for reactNativeVersion. 

I've not tested this in any way yet.

### CODE REVIEW

File: detox/android/detox/build.gradle [hunk 1]
```diff
@@ -10,6 +10,9 @@ def _minSdkVersion = _ext.has('minSdkVersion') ? _ext.minSdkVersion : 18
 def _targetSdkVersion = _ext.has('targetSdkVersion') ? _ext.targetSdkVersion : 25
 def _kotlinVersion = _ext.has('detoxKotlinVersion') ? _ext.detoxKotlinVersion : '1.2.0'
 def _kotlinStdlib = _ext.has('detoxKotlinStdlib') ? _ext.detoxKotlinStdlib : 'kotlin-stdlib-jdk8'
+def _okhttpVersion = _ext.has('okhttpVersion') ? _ext.okhttpVersion : '3.14.1';
+def _okioVersion = _ext.has('okioVersion') ? _ext.okioVersion : '2.2.2';
```
Comment (jgreen210) [inline detox/android/detox/build.gradle]: I've just picked the latest non-alpha/beta version for okhttp and okio. I don't know if they work with kotlin 1.2.0.

File: detox/android/detox/build.gradle [hunk 2]
```diff
@@ -62,11 +65,11 @@ dependencies {
...
 
     // noinspection GradleDynamicVersion
-    compileOnly 'com.facebook.react:react-native:+'
+    compileOnly "com.facebook.react:react-native:${_reactNativeVersion}"
     // noinspection GradleDynamicVersion
-    compileOnly 'com.squareup.okhttp3:okhttp:+'
+    compileOnly "com.squareup.okhttp3:okhttp:${_okhttpVersion}"
     // noinspection GradleDynamicVersion
-    compileOnly 'com.squareup.okio:okio:+'
+    compileOnly "com.squareup.okio:okio:${okioVersion}"
```
Comment (manshubh) [inline detox/android/detox/build.gradle]: I was trying these changes locally, and I guess this should be `_okioVersion `.

Comment (jgreen210) [reply detox/android/detox/build.gradle]: d4709bd55774cbdbf419d1d7ba6d3d9451289761

**Comment by jgreen210** [comment]
Reconfiguring your project as explained in https://github.com/wix/Detox/issues/1380#issuecomment-493955718 means that this PR is not required to solve https://github.com/wix/Detox/issues/1380.

The reactNativeVersion part would still be useful (it's the standard pattern used by react-native libraries now, if all you libraries use this pattern it avoids need to use force in your app's react-native dependency, and even with force a wildcard still makes gradle check all repos to find the latest version making builds unreliable if repos or network fail). That should be done with a separate PR though, since it has nothing to do with this build issue.

**Pull request closed without merging**